run/npc/example-npc2.json

```diff
@@ -25,10 +25,21 @@
             "dialogs": "Hello!\nNice to meet you!",
             "options": {
               "Magic!": "s2",
+              "I've an OTP!": "o1",
               "Bye!": "s4"
             }
           }
         },
+        "o1": {
+            "func": "showDialogAndCheckTOTP",
+            "kwargs": {
+              "dialog": "Give me your key!",
+              "nextState": "s3",
+              "nextStateIncorrect" : "s4",
+              "secret": "TEST",
```

Comment:
I'm not quite familiar with this implementation. Is `secret` a property of the NPC and might be used in other situations (maybe generating the OTP)? Or is it only used in this state function? If it is used in multiple places, we should define it somewhere else once instead of pasting the credential everywhere where it is needed.

Reply:
Secret is the property of the OTP and this implementation aims to allow multiple OTPs to be checked within the same game, and therefore the OTP secret follows the NPC and not in the config, i.e. different NPC can check different OTP.

OTP generation is outside of this code base so consolidation into config doesn't help.